src/ray/object_manager/object_buffer_pool.cc

```diff
@@ -126,6 +126,11 @@ std::pair<const ObjectBufferPool::ChunkInfo &, ray::Status> ObjectBufferPool::Cr
       create_buffer_state_[object_id].chunk_info[chunk_index], ray::Status::OK());
 }
 
+// TODO(rkn): There's something weird about the abort logic here. It only aborts
+// if all chunks are in the available state, but maybe it should abort if all
+// chunks are either available or sealed. Also, if we abort a chunk, then a
+// subsequent create might just recreate the object and leave it only partially
+// filled.
```

Comment:
Yes this part is a bit tricky especially that the receiving code running in separate IO threads